### DISCUSSION TITLE
Having hard time understanding how to use nanojsx Router

### DISCUSSION THREAD

**Discussion by lafusew** [nanojsx/nano#46] [Help / Questions] [answered]
Hey, I'm having hard time understanding how to make the router works on Client Side :(

### What I wanted to achieve 
I would like to have seemless navigation like a client side SPA would. But on the other end having each of my route properly SSR rendered as you can direct. In the template you provide we refresh and GET full new html file only for each page changes.

### The Question I have
Is it possible with NanoJSX to have each route as potentials entry points, but then deliver the javascript needed to handle routes changes on client side ?

### ⚠️ Personal Doubts Warning aha..
My question might be stupid, I started to learn how to use SSR and how Hydratation works using your library, as Nuxt and Next JS were doing to much things behind my back and it just _magically_ worked... Using Nano and your template I can understand properly how its done, thanks for that!

So, If there's something I'm miss understanding or if I'm totally wrong about something, let me now and point me where I should look for :)

### Thanks in advance for your help!

**Comment by yandeu** [comment, marked as answer]
It is possible. But keep in mind the whole point of Nano JSX is to reduce JavaScript. If your app hydrates the whole app, there is not much reduction in size.

I personally never use the router and also had trouble setting it up the way I want.  
(I guess I will have to change the router logic in the future. I don't like the react-like router.)

But, I just made a simple example for you using a simple custom routing method:  
[https://github.com/nanojsx/example1](https://github.com/nanojsx/example1)

Hope it helps :)
